src/io/DumpXTC.cpp

```diff
@@ -0,0 +1,173 @@
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
+
+#include <fstream>
+#include <iomanip>
+#include "DumpXTC.hpp"
+#include "storage/Storage.hpp"
+
+#include "bc/BC.hpp"
+
+#include "analysis/ConfigurationExt.hpp"
+#include "analysis/ConfigurationsExt.hpp"
+
+#include <boost/filesystem.hpp>
+
+using namespace espressopp;
+using namespace espressopp::analysis;
+using namespace std;
+
+namespace espressopp {
+  namespace io {
+    
+    bool DumpXTC::open(const char *mode){
+      
+      if(mode[0] == 'a' && !boost::filesystem::exists(file_name)){
+          fio = open_xtc(file_name.c_str(),"w"); //Opening with mode "a" on a non-existing file leads to error
+      }
+      else{
+          fio = open_xtc(file_name.c_str(),mode); 
+      }
+            
+      return true;
+    }
+
+    void DumpXTC::close(){
+
+      close_xtc(fio);
+      
+      return;
+    }
+
+    void DumpXTC::write(int natoms,  
+                        int step,
+                        float time,
+                        Real3D *box,
+                        Real3D *x,
+                        float prec){
+
+        matrix gmx_box;
+
+        for(int i=0;i<3;i++){
+            for(int j=0;j<3;j++){
+                gmx_box[i][j] = box[i][j];
+            }
+        }
+
+        rvec *gmx_x = new rvec[natoms];
+
+        for(int i=0;i<natoms;i++){
+            gmx_x[i][0] = x[i][0];
+            gmx_x[i][1] = x[i][1];
+            gmx_x[i][2] = x[i][2];
+        }
+        
+        write_xtc(fio,
+                natoms, step, time,
+                gmx_box,gmx_x,prec);
+
+        delete [] gmx_x;
+
+        return;
+    }
+
+
+      
+    void DumpXTC::dump(){
+
+      shared_ptr<System> system = getSystem();
+      ConfigurationsExt conf( system );
+      conf.setUnfolded(unfolded);
+      conf.gather();
+      
+      if( system->comm->rank()==0 ){
+        ConfigurationExtPtr conf_real = conf.back();
+        
+        int num_of_particles = conf_real->getSize();
+       
+        if(this->open("a")){ 
+        
+          ConfigurationExtIterator cei = conf_real-> getIterator();
+          Real3D *box = new Real3D [dim];
+          Real3D *coord = new Real3D [num_of_particles];
+          RealND props;
+          props.setDimension( cei.currentProperties().getDimension() );
+
+          //HACK: Only valid for orthorhombic BC. Will there be anything else in ESPP?
+          Real3D bl=system->bc->getBoxL();
+
+          for(int i=0;i<dim;i++){
+            box[i][0] = 0.;
+            box[i][1] = 0.;
+            box[i][2] = 0.;
+
+            box[i][i] = bl[i]; 
+          }
+
+          for(int i=0;i<num_of_particles;i++){
```

Comment:
maybe here you could also directly create `rvec *gmx_x = new rvec[natoms];` and then you will not need for-loop in line 76; actually `write()` method seems to only rewrite data from one data structure to another one; perhaps this could be reduced to only `dump()` method ?